neural_modelling/src/neuron/current_sources/current_source.h

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2017-2019 The University of Manchester
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//! \dir
+//! \brief Current source implementations
+//! \file
+//! \brief General API of a current source implementation
+#ifndef _CURRENT_SOURCE_H_
+#define _CURRENT_SOURCE_H_
+
+#include <common/neuron-typedefs.h>
+
+#ifdef _CURRENT_SOURCE_DC_H_
+#include "current_source_dc.h"
+#endif
+#ifdef _CURRENT_SOURCE_AC_H_
+#include "current_source_ac.h"
+#endif
+#ifdef _CURRENT_SOURCE_STEP_H_
+#include "current_source_step.h"
+#endif
+#ifdef _CURRENT_SOURCE_NOISY_H_
+#include "current_source_noisy.h"
+#endif
```

Comment:
I think you're right, yes - will check and remove them if that's the case.

Reply:
Actually, it looks like it is necessary, since removing them means that things get bigger in terms of ITCM (there are certain binaries where some of these headers are not included due to there not being enough space available for everything at once).  I don't know whether there might be a less confusing way of doing this...